src/zenml/zen_stores/secrets_stores/azure_secrets_store.py

```diff
@@ -154,39 +180,7 @@ def client(self) -> SecretClient:
         Returns:
             The Azure Key Vault client.
         """
-        if self._client is None:
-            azure_logger = logging.getLogger("azure")
-
-            # Suppress the INFO logging level of the Azure SDK if the
-            # ZenML logging level is WARNING or lower.
-            if logger.level <= logging.WARNING:
-                azure_logger.setLevel(logging.WARNING)
-            else:
-                azure_logger.setLevel(logging.INFO)
-
-            # Initialize the Azure Key Vault client with the
-            # credentials from the configuration, if provided.
-            vault_url = f"https://{self.config.key_vault_name}.vault.azure.net"
-            credential: Union[
-                ClientSecretCredential,
-                DefaultAzureCredential,
-            ]
-            if (
-                self.config.azure_client_id
-                and self.config.azure_tenant_id
-                and self.config.azure_client_secret
-            ):
-                credential = ClientSecretCredential(
-                    tenant_id=self.config.azure_tenant_id.get_secret_value(),
-                    client_id=self.config.azure_client_id.get_secret_value(),
-                    client_secret=self.config.azure_client_secret.get_secret_value(),
-                )
-            else:
-                credential = DefaultAzureCredential()
-            self._client = SecretClient(
-                vault_url=vault_url, credential=credential
-            )
-        return self._client
+        return cast(SecretClient, super().client)
```

Comment:
- Why is there a need to cast this object into `SecretClient`? Don't we already return that type from super().client?

- Why do we not do the same for, say, AWS?

Reply:
This is mainly for mypy and to facilitate auto-complete in the IDE. The `super().client` property returns `Any`. We just cast it here so the Azure specific code knows it is working with an Azure specific client object:

```
    @property
    def client(self) -> Any:
        """Get the secrets store API client.

        Returns:
            The secrets store API client instance.
        """
        # Multiple API calls can be made to this secrets store in the context
        # of different threads. We want to make sure that we only initialize
        # the client once, and then reuse it. We have to use a lock to treat
        # this method as a critical section.
        with self.lock:
            return self._get_client()
```

Reply:
We do the same for GCP:

```
    @property
    def client(self) -> SecretManagerServiceClient:
        """Initialize and return the GCP Secrets Manager client.

        Returns:
            The GCP Secrets Manager client instance.
        """
        return cast(SecretManagerServiceClient, super().client)
```

but not for AWS, (I suppose) because the boto3 library isn't typed and is not covered by mypy (i.e. it uses `Any` for the client type).

Reply:
Yes, if you look in the `pyproject.toml` file, `boto3` is mentioned in the list of python modules that have mypy disabled:

```
[[tool.mypy.overrides]]
module = [
    "tensorflow.*",
    "apache_beam.*",
    "pandas.*",
    ...
    "boto3.*",
    "botocore.*",
    ...
]
```

Reply:
Ah, okay. Got it, thanks!